lib/spack/spack/config.py

```diff
@@ -84,23 +87,47 @@
     'config': spack.schema.config.schema,
 }
 
-#: OrderedDict of config scopes keyed by name.
-#: Later scopes will override earlier scopes.
-config_scopes = OrderedDict()
+#: Builtin paths to configuration files in Spack
+configuration_paths = (
+    # Default configuration scope is the lowest-level scope. These are
+    # versioned with Spack and can be overridden by systems, sites or users
+    ('defaults', os.path.join(spack.paths.etc_path, 'spack', 'defaults')),
+
+    # System configuration is per machine.
+    # No system-level configs should be checked into spack by default
+    ('system', os.path.join(spack.paths.system_etc_path, 'spack')),
+
+    # Site configuration is per spack instance, for sites or projects
+    # No site-level configs should be checked into spack by default.
+    ('site', os.path.join(spack.paths.etc_path, 'spack')),
+
+    # User configuration can override both spack defaults and site config
+    ('user', spack.paths.user_config_path)
```

Comment:
Not sure we should do it in this PR, but I think it's worth swapping `user` scope with `site` scope, and make `site` the default for configuration. Based on my experience using multiple Spack versions at the same time, writing by default to `user` config is very likely to create conflicts if the configuration file format changed in the meanwhile.

Reply:
I think this is good but for another PR.